autolift-core/src/main/scala/autolift/LiftFlatMap.scala

```diff
@@ -1,33 +1,73 @@
 package autolift
 
 
-/**
- * Type class supporting flat mapping a function over an arbitrary nesting of type constructors.
- *
- * @author Owein Reese
- *
- * @tparam Obj The type to be lifted into.
- * @tparam Function The function to be lifted.
- */
-trait LiftFlatMap[Obj, Function] extends DFunction2[Obj, Function]
-
-trait LiftFlatMapSyntax{
-
-	/// Syntax extension providing for a `liftFlatMap` method.
-	implicit class LiftFlatMapOps[F[_], A](fa: F[A]){
-
-		/**
-		 * Automatic lifting and flattening of the contained function `f` such that the application point is dicated by the
-		 * argument and return type of the function.
-		 *
-		 * @param f the function that returns a type with a Monad.
-		 * @tparam B the argument type of the function.
-		 * @tparam C the inner type of the return type of the function.
-		 * @tparam M the higher-kinded type of the return type of the function which has a Monad.
-		 */
-		def liftFlatMap[B, C, M[_]](f: B => M[C])(implicit lift: LiftFlatMap[F[A], B => M[C]]): lift.Out = lift(fa, f)
-	}
+trait LiftFlatMapSemantic { self =>
+  protected type Functor[F[_]]
+  protected type Monad[F[_]]
```

Comment:
So technically, neither Cats nor Scalaz requires a full `Monad`. Instead, Cats requires a `FlatMap` and Scalaz a `Bind`. These are instances which contain a `flatMap` implementation but lack the `pure` of a full on `Monad`. If you add this restriction in, these things won't be able to work with something like a `Map` or any type which could be defined with `Bind`, satisfying those laws but  lacking a `pure`.